utils/node_local_test.py

```diff
@@ -1676,18 +1676,290 @@ def test_pydaos_kv(server, conf):
+        self.conf = conf
+        self.cmd = cmd
+        self.prefix = True
+        self.check_stderr = False
+        self.expected_stdout = None
+        self.use_il = False
+
+    def launch(self):
+        """Run all tests for this command"""
+
+        def _prep(self):
+            rc = self._run_cmd(None)
+            rc.wait()
+            self.expected_stdout = rc.stdout
+            assert not rc.fault_injected
+
+        # Prep what the expected stdout is by running once without faults
+        # enabled.
+        _prep(self)
+
+        print('Expected stdout is')
+        print(self.expected_stdout)
+
+        num_cores = len(os.sched_getaffinity(0))
+
+        if num_cores < 20:
+            max_child = 1
+        else:
+            max_child = int(num_cores / 4 * 3)
+
+        active = []
+        fid = 1
+        max_count = 0
+        finished = False
+
+        # List of fids to re-run under valgrind.
+        to_rerun = []
+
+        fatal_errors = False
+
+        while not finished:
+            if len(active) < max_child and not finished:
+                active.append(self._run_cmd(fid))
+                fid += 1
+
+                if len(active) > max_count:
+                    max_count = len(active)
 
-    Enable the fault injection for the daos binary, injecting
-    allocation failures at different locations.  Keep going until
-    the client runs with no faults injected (about 800 iterations).
+            # Now complete as many as have finished.
+            while active and active[0].has_finished():
+                ret = active.pop(0)
+                print(ret)
+                if ret.returncode < 0:
+                    fatal_errors = True
+                    to_rerun.append(ret.loc)
 
-    Disable valgrind for this test as it takes a long time to run
-    with valgrind enabled, use purely the log analysis to find issues.
+                if not ret.fault_injected:
+                    print('Fault injection did not trigger, stopping')
+                    finished = True
 
-    Ignore new error messages containing the numeric value of -DER_NOMEM
-    but warn on all other warnings generated.
+        print('Completed, fid {}'.format(fid))
+        print('Max in flight {}'.format(max_count))
+
+        for fid in to_rerun:
+            rerun = self._run_cmd(fid, valgrind=True)
+            print(rerun)
+            rerun.wait()
+
+        return fatal_errors
+
+    def _run_cmd(self,
+                 loc,
+                 valgrind=False):
+        """Run the test with FI enabled
+        """
+
+        cmd_env = get_base_env()
+
+        if self.use_il:
+            cmd_env['LD_PRELOAD'] = os.path.join(self.conf['PREFIX'],
+                                                 'lib64', 'libioil.so')
+
+        cmd_env['DAOS_AGENT_DRPC_DIR'] = self.conf.agent_dir
+
+        aftf = AllocFailTestRun(self, self.cmd, cmd_env, loc)
+        if valgrind:
+            aftf.vh = ValgrindHelper()
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'conf' in constructor call